R/derive_vars_transposed.R

```diff
@@ -84,19 +93,46 @@ derive_vars_transposed <- function(dataset,
                                    by_vars,
                                    key_var,
                                    value_var,
-                                   filter = NULL) {
+                                   filter = NULL,
+                                   relationship = NULL) {
   key_var <- assert_symbol(enexpr(key_var))
   value_var <- assert_symbol(enexpr(value_var))
   filter <- assert_filter_cond(enexpr(filter), optional = TRUE)
   assert_vars(by_vars)
   assert_data_frame(dataset, required_vars = replace_values_by_names(by_vars))
   assert_data_frame(dataset_merge, required_vars = expr_c(by_vars, key_var, value_var))
+  relationship <- assert_character_scalar(
+    relationship,
+    values = c("one-to-one", "many-to-one"),
```

Comment:
This function is an exception because it may add records (like in the example). Thus we should allow also `"one-to-many"` and `"many-to-many"`.

Reply:
Ah right ok, I will add those back in to that function then. Apologies!